src/util/adopt-css.js

```diff
@@ -3,12 +3,19 @@ let styles = new WeakMap();
 /**
  * Adopt CSS into a document or shadow root.
  * @param {string} css - The CSS to adopt.
- * @param {Document|ShadowRoot} [root=globalThis.document] - The document or shadow root to adopt the CSS into.
+ * @param {Object|Document|ShadowRoot} [options] - The options object. If a document or shadow root is provided, it will be used as the root.
+ * @param {Document|ShadowRoot} [options.root=globalThis.document] - The document or shadow root to adopt the CSS into.
+ * @param {boolean} [options.isolated=false] - Whether to create a new `<style>` element if adopted style sheets are not supported. Useful if you plan to modify or remove it later.
+ * @returns {CSSStyleSheet} The adopted style sheet.
  */
-export default function adoptCSS (css, root = globalThis.document) {
+export function adoptCSS (css, options = {}) {
+	if (options?.constructor.name !== "Object") {
+		options = { root: options };
+	}
+
 	// Ensure root is always a document
-	root = root.ownerDocument ?? root;
-	let window = root.defaultView;
+	let root = options.root.getRootNode?.() ?? options.root ?? globalThis.document;
```

Comment:
```suggestion
	let root = options.root?.getRootNode?.() ?? options.root ?? globalThis.document;
```

We should probably use the optional chaining a step earlier. What if `root` wasn't passed as a property of the `options` object?